### DISCUSSION TITLE
New effects not taking on flash.

### DISCUSSION THREAD

**Discussion by cnf** [PlummersSoftwareLLC/NightDriverStrip#271] [General]
Hi,

I made a new config in `globals.h` and `effect.cpp`, but  when I compile and flash it, The effect that were there from a previous flash don't change. Settings like `DEFAULT_EFFECT_INTERVAL` and `LED_PIN0` and `TOGGLE_BUTTON_1` all get picked up, so the flash seems successful, I just seem unable to get any new effects on the board... Am I missing something here?

Thanks!

**Comment by davepl** [comment]
I think it’s likely remembering its old config.  Try running 

epsytool.py —erase_flash

Then refresh the data and code, and it should work properly!

Rutger may also be able to tell you how to reset the config with Postman.

-Dave

**Comment by cnf** [comment]
That seems to have done the trick, indeed!

As you mentioned Postman, I poked around a bit more and saw it is actually documented [here](https://github.com/PlummersSoftwareLLC/NightDriverStrip/blob/main/REST_API.md#reset-configuration-andor-device). 

Maybe an idea to add to the README, or make a pio target for this?

Thanks for the help @davepl !

**Reply by rbergen** [reply]
Hi @cnf, there is actually a link to that page in the README. We've reached the point there's so much to cover in documentation that cramming it all in one page is no longer practical. 

What we actually need is a UI with which all API capabilities can be used in an easy manner. This is exactly why Dave did a call-out for web developers specifically in a recent video. 

All that said, I'll see if I can add a brief "(Re)setting settings" section to the README with another pointer to the API page. That may make it a bit easier to find.

**Reply by cnf** [reply]
Hey @rbergen, once I knew what I was looking for with Dave's help, I found it easy enough. I just didn't know I needed it 😅 

I saw a mention of effects.cfg in the serial log once in a while, but it took me a while to figure out that was what was needed.

So, for me at least, a mention that you need to reset these if you change your effects was the missing link!

Thanks for the help!

**Comment by rbergen** [comment]
PR #282 has been merged to add some specific notes concerning this to README.md.

**Reply by cnf** [reply]
Perfect, thanks a lot @rbergen !

**Comment by rbergen** [comment]
Just for completeness: I've just opened PR #287, which automatically adds any new effects that are available for a project to the effect list that's loaded from JSON. The README will be updated accordingly.
